### ISSUE TITLE
support variable arguments in directive definition

### ISSUE THREAD

**Issue by ronjunevaldoz** [ExpediaGroup/graphql-kotlin#1404]
Below directive throws an error

`com.expediagroup.graphql.generator.exceptions.TypeNotSupportedException: Cannot convert kotlin.Array<out kotlin.String> since it is not a valid GraphQL type or outside the supported packages`

```
@GraphQLDirective(
    name = "auth",
    description = "Authenticated user"
)
annotation class AuthDirective(vararg val role: String)
```

**Comment by dariuszkuc** [comment]
Hi 👋 
Thanks for raising this. Indeed this is a valid use case that we currently don't support. Looks like we broke it when we fixed the duplicate serialization issue #1379 (which dropped support for arrays).

**Comment by ronjunevaldoz** [comment]
Hello @dariuszkuc :grinning:

Thank you for your very responsive support. I guess I'm going to use a single argument for now.
